horovod/common/operations.cc

```diff
@@ -1956,36 +1989,6 @@ void horovod_shutdown() {
     horovod_global.initialize_flag.clear();
     horovod_global.shut_down = false;
   }
-
-  if (horovod_global.mpi_comm != MPI_COMM_NULL &&
-      horovod_global.mpi_comm != MPI_COMM_WORLD) {
-    MPI_Comm_free(&horovod_global.mpi_comm);
```

Comment:
@vloncar, do you remember why this logic ended up in `horovod_shutdown()` instead of `BackgroundThread` code?

Reply:
@alsrgv The original idea was to to move the `MPI_Finalize()` out of the `BackgroundThread` so that if Horovod is embedded into a larger MPI application it won't stop everything once it finishes thus allowing for restarts (a repeat of `hvd.init()` & `hvd.shutdown()` sequence in a larger MPI app). Later refinements have added the ability for Horovod to figure out if it needs to call `MPI_Finalize()` or just free up internal communicators, so yeah, it doesn't really need to be done in `shutdown()`. I vaguely remember there being some issues with MPI finalizing in one background thread if background threads on other ranks are still communicating, but I can't reproduce those issues with this. I have tested your change in our apps, and it works as expected, so LGTM.